presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -1612,6 +1612,10 @@ private static boolean isDistinct(SqlBaseParser.SetQuantifierContext setQuantifi
     private static String getType(SqlBaseParser.TypeContext type)
     {
         if (type.baseType() != null) {
+            if (type.baseType().DOUBLE_PRECISION() != null) {
+                // Make DOUBLE PRECISION alias for DOUBLE.
+                return "DOUBLE";
```

Comment:
Instead of returning `"DOUBLE"`, set `signature` to `"DOUBLE"`. `DOUBLE` shouldn't have type parameters. But if there are, we shouldn't ignore them. They need to kept around so that analyzer (which happens after parser, and before planner) can look at the invalid combination, and fail the query properly.